Utilities/bootstrap

```diff
@@ -196,8 +202,12 @@ def parse_build_args(args):
     args.clang_path = get_clang_path(args)
     args.cmake_path = get_cmake_path(args)
     args.ninja_path = get_ninja_path(args)
-    if args.cross_compile_hosts: # Use XCBuild target directory when building for multiple arches.
-        args.target_dir = os.path.join(args.build_dir, "apple/Products")
+    if args.cross_compile_hosts:
+        if "macosx-arm64" in args.cross_compile_hosts:
```

Comment:
I don't think that this is correct; there are a number of targets that may be used, can you just invert the condition here instead?

```
if args.cross_compile_hosts:
  if re.match('android-', args.cross_compile_hosts):
    args.target_dir = os.path.join(args.build_dir, get_build_targets(args, cross_compile=True))
  else:
    args.target_dir = os.path.join(args.build_dir, 'apple/Products')
```

Yes, it isn't great, but leaves the behaviour the same while adding support for your use case.

Reply:
No, this is right, [see the similar logic at the end of the script](https://github.com/apple/swift-package-manager/blob/6af862d3a7efb3108334be0d3e3e33641c91f267/Utilities/bootstrap#L805). Basically, Mishal only added support for cross-compiling SPM for macOS arm64 and he explicitly spelled that out below, but didn't bother here: I went ahead and made it explicit here too, now that I added Android also.

Reply:
WFM